test/index.js

```diff
@@ -353,6 +363,14 @@ describe('stream', function() {
       s(1)(2);
       assert.deepEqual(result, [1, 2]);
     });
+    it('is only invoked when the stream changes to a Right', function() {
+      var s = flyd.stream();
+      var result = [];
+      var f = function(v) { result.push(v); };
+      flyd.on(f, s);
+      s(Either.Right(1))(Either.Left(-1))(Either.Right(2));
```

Comment:
So at the point where you push a left into the stream, is the Stream a left? If so, it should not be able to go back to being a right at any point -- it should be as if the stream has ended in a way.